compiler/ballerina-lang/src/main/java/io/ballerina/projects/ResolvedPackageDependency.java

```diff
@@ -61,4 +61,9 @@ public boolean equals(Object o) {
     public int hashCode() {
         return super.hashCode();
     }
+
+    @Override
+    public int compareTo(ResolvedPackageDependency resolvedPackageDependency) {
+        return this.packageId().toString().compareTo(resolvedPackageDependency.packageId().toString());
```

Comment:
We should use the org and the name from the `resolvedPackage.descriptor()` for the comparison AFAIU